src/index.js

```diff
@@ -20,10 +20,185 @@
 
 import SpanContext from './span_context';
 import Span from './span';
+import ConstSampler from './samplers/const_sampler';
+import InMemoryReporter from './reporters/in_memory_reporter';
+import NoopReporter from './reporters/noop_reporter';
+import ProbabilisticSampler from './samplers/probabilistic_sampler';
+import RatelimitingSampler from './samplers/ratelimiting_sampler';
+import RemoteReporter from './reporters/remote_reporter';
+import RemoteSampler from './samplers/remote_sampler';
 import Tracer from './tracer';
+import UDPSender from './reporters/udp_sender';
+import {Validator} from 'jsonschema';
+
+
+let configSchema = {
+    'type': 'object',
+    'required': ['jaeger'],
+    'properties': {
+        'jaeger': {
+            'type': 'object',
+            '$ref': '/jaeger'
+        }
+    },
+};
+
+let constSamplerSchema = {
+    'id': '/const',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'boolean' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let probabilisticSamplerSchema = {
+    'id': '/probabilistic',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'number' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let ratelimitingSamplerSchema = {
+    'id': '/ratelimiting',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'number' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let remoteSamplerSchema = {
+    'id': '/remote',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'string' },
+        'managerHostPort': {'type': 'string'}
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let jaegerSchema = {
+    'id': '/jaeger',
+    'type': 'object',
+    'required': ['serviceName'],
```

Comment:
My problem with this is that all new instrumentation will use "initTracer" and pass in a serviceName that can be incorrect, or fall out of sync from what they have in their yaml file.  So it makes sense for it to be in the config.  The internal repo can still read service name from an env variable, and modify the config it receives.  Basically since most apps determine there own serviceName (which can be wrong) and keep it in yaml files.  I think it makes sense to keep serviceName in the config.